templates/cli/lib/commands/pull.js.twig

```diff
@@ -26,142 +23,77 @@ const pull = new Command("pull")
+        functions = (await paginate(functionsList, { parseOutput: false }, 100, 'functions')).functions;
     }
 
-    const functionId = answers.id === 'unique()' ? ID.unique() : answers.id;
-    const functionDir = path.join(functionFolder, functionId);
+    const answers = await inquirer.prompt(questions);
 
-    if (fs.existsSync(functionDir)) {
-        throw new Error(`( ${functionId} ) already exists in the current directory. Please choose another name.`);
-    }
+    const overridingLocalChanges = yes ?? answers.override.toLowerCase() === "yes";
+    const selectedFunctions = functions.length === 0 ? answers.functions : functions;
 
-    if (!answers.runtime.entrypoint) {
-        log(`Entrypoint for this runtime not found. You will be asked to configure entrypoint when you first push the function.`);
-    }
+    for (let func of selectedFunctions) {
+        const functionExistLocally = localFunctions.find((localFunc) => localFunc['$id'] === func['$id']) !== undefined;
 
-    if (!answers.runtime.commands) {
-        log(`Installation command for this runtime not found. You will be asked to configure the install command when you first push the function.`);
-    }
+        if (!overridingLocalChanges && functionExistLocally) {
+            log(`Skipping locally found implementation of ${func['name']}`)
+            continue;
+        }
+        if (functionExistLocally) {
+            localConfig.updateFunction(func['$id'], func);
+        } else {
+            func['path'] = `functions/${func['$id']}`;
+            localConfig.addFunction(func);
+        }
 
-    let response = await functionsCreate({
-        functionId,
-        name: answers.name,
-        runtime: answers.runtime.id,
-        entrypoint: answers.runtime.entrypoint || '',
-        commands: answers.runtime.commands || '',
-        parseOutput: false
-    })
+        const localFunction = localFunctions.find((localFunc) => localFunc['$id'] === func['$id']);
 
-    fs.mkdirSync(functionDir, "777");
+        if (localFunction['deployment'] === '') {
+            continue
+        }
 
-    let gitInitCommands = "git clone -b v3 --single-branch --depth 1 --sparse https://github.com/{{ sdk.gitUserName }}/functions-starter ."; // depth prevents fetching older commits reducing the amount fetched
+        const compressedFileName = `${+new Date()}.tar.gz`
 
-    let gitPullCommands = `git sparse-checkout add ${answers.runtime.id}`;
+        await functionsDownloadDeployment({
+            functionId: func['$id'],
+            deploymentId: func['deployment'],
+            destination: compressedFileName,
+            overrideForCli: true,
+            parseOutput: false
+        })
 
-    /* Force use CMD as powershell does not support && */
-    if (process.platform === 'win32') {
-        gitInitCommands = 'cmd /c "' + gitInitCommands + '"';
-        gitPullCommands = 'cmd /c "' + gitPullCommands + '"';
-    }
+        tar.extract({
```

Comment:
Not sure if relevant, but we not only tar, we also zip (tar.gz). We may need to unzip too, or provide some parameter to tar to tell it to unzip too.

If you did QA and it worked well, you can ignore this comment